uk.co.compucorp.civicrm.tasksassignments/tasksassignments.php

```diff
@@ -173,25 +173,30 @@ function tasksassignments_civicrm_pageRun($page) {
 function tasksassignments_civicrm_tabs(&$tabs) {
     CRM_Tasksassignments_Page_Tasksassignments::registerScripts();
 
-    $tabs[] = Array(
+    if (CRM_Core_Permission::check('access Tasks and Assignments')) {
+      $tabs[] = Array(
         'id'        => 'civitasks',
         'url'       => CRM_Utils_System::url('civicrm/contact/view/tasks'),
         'title'     => ts('Tasks'),
         'weight'    => 40
-    );
+      );
+    }
 
     $documentsTab = civicrm_api3('TASettings', 'get', array(
-        'fields' => 'documents_tab',
+      'fields' => 'documents_tab',
     ));
-    if ($documentsTab['values']['documents_tab']['value']) {
-        $tabs[] = Array(
-            'id'        => 'cividocuments',
-            'url'       => CRM_Utils_System::url('civicrm/contact/view/documents'),
-            'title'     => ts('Documents'),
-            'weight'    => 50
-        );
-    }
 
+    if (
```

Comment:
a single check for permission is enough ... so you could change the code to something like :

if (CRM_Core_Permission::check('access Tasks and Assignments')) {
      $tabs[] = Array(
        'id'        => 'civitasks',
        'url'       => CRM_Utils_System::url('civicrm/contact/view/tasks'),
        'title'     => ts('Tasks'),
        'weight'    => 40
      );
    $documentsTab = civicrm_api3('TASettings', 'get', array(
      'fields' => 'documents_tab',
    ));

if ( $documentsTab['values']['documents_tab']['value']) {
      $tabs[] = Array(
        'id'        => 'cividocuments',
        'url'       => CRM_Utils_System::url('civicrm/contact/view/documents'),
        'title'     => ts('Documents'),
        'weight'    => 50
      );
    }
    }